docs/tutorial/expressjs.rst

```diff
@@ -0,0 +1,430 @@
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the ExpressJS
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:curl-expressjs-rock]
+    :end-before: [docs:curl-expressjs-rock-end]
+    :dedent: 2
+
+The ExpressJS application should again respond with the ``Welcome to Express``
+HTML.
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the ExpressJS rock, we can always get the application logs via
+:ref:`pebble_explanation_page`:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, Pebble will give us the logs for the
+``expressjs`` service running inside the container.
+We should expect to see something similar to this:
+
+.. terminal::
+
+    app@0.0.0 start
+    node ./bin/www
+    GET / 200 62.934 ms - 170
+
+We can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` +
+:kbd:`C`.
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a ExpressJS application! This concludes
+the first part of this tutorial, so we'll stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+Chisel the rock
+===============
+
+This is an optional but recommended step, especially if we're looking to
+deploy the rock into a production environment. With :ref:`chisel_explanation`
+we can produce lean and production-ready rocks by getting rid of all the
+contents that are not needed for the ExpressJS application to run. This results
+in a much smaller rock with a reduced attack surface.
+
+.. note::
+    It is recommended to run chiselled images in production. For development,
+    we may prefer non-chiselled images as they will include additional
+    development tooling (such as for debugging).
+
+The first step towards chiselling the rock is to ensure we are using a
+``bare`` :ref:`base <bases_explanation>`.
+In ``rockcraft.yaml``, change the ``base`` to ``bare`` and add
+``build-base: ubuntu@24.04``:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:change-base]
+    :end-before: [docs:change-base-end]
+    :dedent: 2
+
+.. note::
+    The ``sed`` command replaces the current ``base`` in ``rockcraft.yaml`` with
+    the ``bare`` base. Note that ``build-base`` is also required when using the
+    ``bare`` base.
+
+So that we can compare the size after chiselling, open the ``rockcraft.yaml``
+file and change the ``version`` (e.g. to ``0.1-chiselled``). Pack the rock with
+the new ``bare`` :ref:`base <bases_explanation>`:
```

Comment:
```suggestion
the new bare base:
```

Reply:
Bare base tutorial has been removed in favor of matching the style in `go` tutorial.

Reply:
(the default starts with bare base)